pkg/api/multiarchbuildconfig/v1/types.go

```diff
@@ -66,14 +66,16 @@ func UpdateMultiArchBuildConfig(ctx context.Context, logger *logrus.Entry, clien
 	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
 		mabc := &MultiArchBuildConfig{}
 		if err := client.Get(ctx, namespacedName, mabc); err != nil {
+			logger.Error("Failed to get the MultiArchBuildConfig while updating")
```

Comment:
But we are already returning the error. You need to log it on that layer.

Reply:
You mean outside the `RetryOnConflict` every time `UpdateMultiArchBuildConfig` is invoked?

Reply:
This will retry only on conflicts. Technically we already return the error. You can add the log there.

Reply:
Something like this?
```diff
if err := v1.UpdateMultiArchBuildConfig(ctx, logger, r.client, ctrlruntimeclient.ObjectKey{Namespace: mabc.Namespace, Name: mabc.Name}, mutateFn); err != nil {
+   logger.Errorf("Failed to update the mabc: %s", err)
    return fmt.Errorf("failed to update the MultiArchBuildConfig %s/%s: %w", mabc.Namespace, mabc.Name, err)
}
```

Reply:
We are returning that error too.